### PR TITLE
Build logs

### PR DESCRIPTION
**Pull request by nzoschke**

### CODE REVIEW

**Comment by nzoschke** [comment]
`GET /apps/{app}/builds/{build}/logs` returns the build logs, which are periodically updated (currently every 2 seconds) during a bulid.

@mattmanning with this you should be able to refresh the UI every to seconds to simulate real-time build logs with this.

**Comment by nzoschke** [comment]
I am investigating a `/apps/{app}/builds/{build}/logs/stream` streaming websocket handler next...

**Comment by nzoschke** [comment]
The streaming implementation looks like it's working from `wscat`!

Both the writing and reading are my first application of the Ticker. I'd be happy to get some feedback on how bad my go code is here :)

**Comment by nzoschke** [comment]
```
$ wscat -c ws://192.168.59.103:5000/apps/app-flaming-sword-16/builds/BDWOEMHAUYM/logs/stream

connected (press CTRL+C to quit)

< RUNNING: git clone https://github.com/convox-examples/sinatra.git /tmp/repo027677254/clone
Cloning into '/tmp/repo027677254/clone'...
RUNNING: /usr/local/bin/git-restore-mtime .
...
```

**Comment by nzoschke** [comment]
```
$ curl localhost:5000/apps/app-flaming-sword-16/builds/BRBDBOQZLMI/logs
RUNNING: git clone https://github.com/convox-examples/sinatra.git /tmp/repo796982026/clone
Cloning into '/tmp/repo796982026/clone'...
RUNNING: /usr/local/bin/git-restore-mtime .
RUNNING: docker-compose -p axvlbzgba build
redis uses an image, skipping
...
```

**Comment by nzoschke** [comment]
You can also update the Status and Logs in dynamodb directly to test things. It's pretty cool to put the state back to 'building' and manually enter some additional log lines.

**Comment by ddollar** [comment]
This is looking good

**Comment by mattmanning** [comment]
Front end changes:
- When the build button is clicked switch to builds tab and open build logs modal.
- Live build log with autoscrolling until user scroll input
- Builds table auto-reloads every 5 seconds so you can quickly see when a build is finished

**Comment by ddollar** [comment]
:+1:

File: views/app.tmpl [hunk 1]
```diff
@@ -296,6 +300,44 @@
...
+				}
+				ticker = window.setInterval(function() {
+					$.ajax({ url:'/apps/{{ .App.Name }}/builds/'+build_id+'/logs' }).done(function(data) {
+						modal.find('pre').text(data);
+					});
+					if (autoscroll) {
+						modal_scroll.scrollTop(modal_scroll.prop("scrollHeight"));
+						scroll_position = modal_scroll.scrollTop();
+					}
+				}, 500);
```
Comment (nzoschke) [inline views/app.tmpl]: 500ms is pretty aggressive. The backend saves new logs every 2 seconds.

**Comment by nzoschke** [comment]
Does this have something to check if a build is successful? In testing it kept hammering the webserver even though no new logs will ever come through.

**Comment by nzoschke** [comment]
The above question will probably be improved via the 'Build log modal should have a “Promote” button at the bottom.' task

**Comment by mattmanning** [comment]
@nzoschke Good points. I think we can use the build status to decide whether to keep polling for logs. Just trying to get the quick and dirty working first.

**Pull request merged**